File: Sep19/Lecture8Examples/Example32/MainActivity.java
package com.varunshankar.example32;

import android.app.job.JobInfo;
import android.app.job.JobScheduler;
import android.content.ComponentName;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.RadioGroup;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private JobScheduler mScheduler;
    private static final int JOB_ID = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mScheduler = (JobScheduler) getSystemService(JOB_SCHEDULER_SERVICE);
    }

    public void scheduleJob(View view){
        RadioGroup networkOptions = (RadioGroup) findViewById(R.id.networkOptions);
        int selectedNetworkID = networkOptions.getCheckedRadioButtonId();

        int selectedNetworkOption = JobInfo.NETWORK_TYPE_NONE;

        switch(selectedNetworkID){
            case R.id.noNetwork:
                selectedNetworkOption = JobInfo.NETWORK_TYPE_NONE;
                break;
            case R.id.anyNetwork:
                selectedNetworkOption = JobInfo.NETWORK_TYPE_ANY;
                break;
            case R.id.wifiNetwork:
                selectedNetworkOption = JobInfo.NETWORK_TYPE_UNMETERED;
                break;
        }

        ComponentName serviceName = new ComponentName(getPackageName(),NotificationJobService.class.getName());
        JobInfo.Builder builder = new JobInfo.Builder(JOB_ID,serviceName);
        builder.setRequiredNetworkType(selectedNetworkOption);
        builder.setRequiresCharging(false);

        JobInfo myJobInfo = builder.build();
        mScheduler.schedule(myJobInfo);
        Toast.makeText(this,"Job was scheduled!",Toast.LENGTH_SHORT).show();
    }

    public void cancelJobs(View view){
        if (mScheduler!=null){
            mScheduler.cancelAll();
            mScheduler = null;
            Toast.makeText(this, "Jobs Canceled", Toast.LENGTH_SHORT).show();
        }
    }

}

File: Sep19/Example32/app/src/main/java/com/varunshankar/example32/MainActivity.kt
package com.varunshankar.example32

import androidx.appcompat.app.AppCompatActivity
import android.app.job.JobScheduler
import android.os.Bundle
import android.widget.RadioGroup
import android.app.job.JobInfo
import android.content.ComponentName
import android.view.View
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    private var mScheduler: JobScheduler? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mScheduler = getSystemService(JOB_SCHEDULER_SERVICE) as JobScheduler
    }

    fun scheduleJob(view: View?) {
        val networkOptions = findViewById<View>(R.id.networkOptions) as RadioGroup
        val selectedNetworkID = networkOptions.checkedRadioButtonId
        var selectedNetworkOption = JobInfo.NETWORK_TYPE_NONE
        when (selectedNetworkID) {
            R.id.noNetwork -> selectedNetworkOption = JobInfo.NETWORK_TYPE_NONE
            R.id.anyNetwork -> selectedNetworkOption = JobInfo.NETWORK_TYPE_ANY
            R.id.wifiNetwork -> selectedNetworkOption = JobInfo.NETWORK_TYPE_UNMETERED
        }
        val serviceName = ComponentName(packageName, NotificationJobService::class.java.name)
        val builder = JobInfo.Builder(JOB_ID, serviceName)
        builder.setRequiredNetworkType(selectedNetworkOption)
        builder.setRequiresCharging(false)
        val myJobInfo = builder.build()
        mScheduler!!.schedule(myJobInfo)
        Toast.makeText(this, "Job was scheduled!", Toast.LENGTH_SHORT).show()
    }

    fun cancelJobs(view: View?) {
        if (mScheduler != null) {
            mScheduler!!.cancelAll()
            mScheduler = null
            Toast.makeText(this, "Jobs Canceled", Toast.LENGTH_SHORT).show()
        }
    }

    companion object {
        private const val JOB_ID = 0
    }
}